README.md

```diff
@@ -145,7 +166,7 @@ code, include a script or makefile that will do the job.
 
 ## A2.5
 
-«replace this with your answer»
+node A25.js
```

Comment:
First, thank you for giving me both positive and negative test, and for making it clear which are which. The only thing I'd ask of you in future is to make the program itself know which should succeed and fail, so that it can automatically report bugs when an actual result isn't the same as the expected.

Unfortunately, I do have a problem with your FSM. The question asked you to express the table in your language and then write an interpreter that used it. That's ridiculously easy in JavaScript:

~~~ js
states = {
  0: { "the": 1 },
  1: { "lazy": 1, "smelly": 1, "dog": 2, "cat": 2 },
  2: { "ate": 3, "ran": 3 },
  3: { "slowly": 4, "noisily": 4, "EOI": END },
  4: { "EOI": END }
};

function getNextState(currentState, nextWord) {
  return currentState && states[currentState][nextWord]
}

function stateMachine(sentence) {
  return sentence.reduce(getNextState, 0)
}
~~~

Your code doesn't actually use states, except as documentation. 

As a result it doesn't actually parse according to your table: it doesn't allow repeating adjectives

# -8